test/FullRange.t.sol

```diff
@@ -0,0 +1,453 @@
+        manager.initialize(key, SQRT_RATIO_1_1);
+
+        uint256 prevBalance0 = MockERC20(token0).balanceOf(address(this));
+        uint256 prevBalance1 = MockERC20(token1).balanceOf(address(this));
+
+        address token0Addr = address(token0);
+        address token1Addr = address(token1);
+
+        fullRange.addLiquidity(token0Addr, token1Addr, 3000, 10 ether, 10 ether, address(this), MAX_DEADLINE);
+
+        (, address liquidityToken) = fullRange.poolInfo(id);
+
+        assertEq(UniswapV4ERC20(liquidityToken).balanceOf(address(this)), 10 ether - 1000);
+
+        assertEq(MockERC20(token0).balanceOf(address(this)), prevBalance0 - 10 ether);
+        assertEq(MockERC20(token1).balanceOf(address(this)), prevBalance1 - 10 ether);
+
+        UniswapV4ERC20(liquidityToken).approve(address(fullRange), type(uint256).max);
+
+        snapStart("remove liquidity");
+        fullRange.removeLiquidity(token0Addr, token1Addr, 3000, 1 ether, MAX_DEADLINE);
+        snapEnd();
+
+        (bool owed,) = fullRange.poolInfo(id);
+
+        assertEq(UniswapV4ERC20(liquidityToken).balanceOf(address(this)), 9 ether - 1000);
+        assertEq(MockERC20(token0).balanceOf(address(this)), prevBalance0 - 9 ether - 1);
+        assertEq(MockERC20(token1).balanceOf(address(this)), prevBalance1 - 9 ether - 1);
+        assertEq(owed, false);
+    }
+
+    function testInitialRemoveLiquidityFuzz(uint256 amount) public {
+        manager.initialize(key, SQRT_RATIO_1_1);
+
+        fullRange.addLiquidity(
+            address(token0), address(token1), 3000, 1000 ether, 1000 ether, address(this), MAX_DEADLINE
+        );
+
+        (, address liquidityToken) = fullRange.poolInfo(id);
+
+        UniswapV4ERC20(liquidityToken).approve(address(fullRange), type(uint256).max);
+
+        if (amount >= 1000 ether - 1000) {
+            vm.expectRevert();
+            fullRange.removeLiquidity(address(token0), address(token1), 3000, amount, MAX_DEADLINE);
+        } else {
+            fullRange.removeLiquidity(address(token0), address(token1), 3000, amount, MAX_DEADLINE);
+
+            (bool owed,) = fullRange.poolInfo(id);
+            assertEq(owed, false);
+        }
+    }
+
+    function testRemoveLiquidityFailsIfNoPool() public {
+        vm.expectRevert(FullRange.PoolNotInitialized.selector);
+        fullRange.removeLiquidity(address(token0), address(token1), 0, 10 ether, MAX_DEADLINE);
+    }
+
+    function testRemoveLiquidityFailsIfNoLiquidity() public {
+        manager.initialize(key, SQRT_RATIO_1_1);
+
+        (, address liquidityToken) = fullRange.poolInfo(id);
+        UniswapV4ERC20(liquidityToken).approve(address(fullRange), type(uint256).max);
+
+        vm.expectRevert(); // Insufficient balance error from ERC20 contract
+        fullRange.removeLiquidity(address(token0), address(token1), 3000, 10 ether, MAX_DEADLINE);
+    }
+
+    function testRemoveLiquiditySucceedsWithPartialAndFee() public {
```

Comment:
not sure I understand "withPartialAndFee"

Reply:
would think that "withFee" means they're taking out their liquidity + owed swap fees?